pkg/kubectl/cmd/util/factory.go

```diff
@@ -254,8 +254,8 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			}
 			return kubectl.ReaperFor(mapping.Kind, client, expClient)
 		},
-		Validator: func() (validation.Schema, error) {
-			if flags.Lookup("validate").Value.String() == "true" {
+		Validator: func(validate bool) (validation.Schema, error) {
```

Comment:
I vastly prefer this to looking up the flag from inside the function, but why would something with `validate == false` ever call the `Validator()` function?